app/model/entity/ShadowAssignment.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace App\Model\Entity;
+
+use DateTime;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\Selectable;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Criteria;
+use Kdyby\Doctrine\MagicAccessors\MagicAccessors;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * @ORM\Entity
+ * @Gedmo\SoftDeleteable(fieldName="deletedAt", timeAware=false)
+ *
+ * @method string getId()
+ * @method DateTime getCreatedAt()
+ * @method int getMaxPoints()
+ * @method Collection getShadowAssignmentEvaluations()
+ * @method setIsPublic(bool $public)
+ * @method setIsBonus(bool $bonus)
+ * @method setMaxPoints(int $points)
+ */
+class ShadowAssignment implements IAssignment
+{
+  use MagicAccessors;
+  use UpdateableEntity;
+  use DeleteableEntity;
+  use VersionableEntity;
+
+  private function __construct(
+    int $maxPoints,
+    Group $group,
+    bool $isPublic,
+    bool $isBonus = false
+  ) {
+    $this->group = $group;
+    $this->maxPoints = $maxPoints;
+    $this->shadowAssignmentEvaluations = new ArrayCollection;
+    $this->isPublic = $isPublic;
+    $this->localizedTexts = new ArrayCollection();
+    $this->version = 1;
+    $this->isBonus = $isBonus;
+    $this->createdAt = new \DateTime;
+    $this->updatedAt = new \DateTime;
+  }
+
+  public static function assignToGroup(Group $group, $isPublic = false) {
```

Comment:
I'm not sure I like this name - it kinda makes sense for exercises, but here it's just a weird constructor. Couldn't we just add `$group->addShadowAssignment()` to the constructor and remove this method alltogether?

Reply:
There are some default values in this factory function handed over to constructor. I would preserve this function and name it more for your likeliness... `createInGroup` maybe?

Reply:
Yeah, that's better.